tests/robustness/failpoint/gofail.go

```diff
@@ -189,3 +191,41 @@ func (f killAndGofailSleep) Available(config e2e.EtcdProcessClusterConfig, membe
 	}
 	return memberFailpoints.Available(f.failpoint)
 }
+
+type gofailSleepAndDeactivate struct {
+	failpoint string
+	time      time.Duration
+}
+
+func (f gofailSleepAndDeactivate) Inject(ctx context.Context, t *testing.T, lg *zap.Logger, clus *e2e.EtcdProcessCluster) error {
+	member := clus.Procs[rand.Int()%len(clus.Procs)]
+	select {
+	case <-ctx.Done():
+		return ctx.Err()
+	default:
+	}
+	lg.Info("Setting up gofailpoint", zap.String("failpoint", f.Name()))
+	err := member.Failpoints().SetupHTTP(ctx, f.failpoint, fmt.Sprintf(`sleep(%q)`, f.time))
+	if err != nil {
+		lg.Info("goFailpoint setup failed", zap.String("failpoint", f.Name()), zap.Error(err))
+	}
+	time.Sleep(f.time)
```

Comment:
How can you be sure that failpoint was really executed?

Reply:
Would you be interested in looking into adding a check that etcd was really executed the failpoint sleep?

Required change had no progress for some time https://github.com/etcd-io/gofail/pull/47

For example see https://github.com/etcd-io/etcd/pull/16776

Reply:
I have tested it. Without deactivating failpoint, the qps dropped down since the simulate traffic on failpoint injected member didn't finish on time, especially when I set a longer time interval to sleep. 

For https://github.com/etcd-io/gofail/pull/47 Sure, I'll take a look. It would be useful to have such functionality rather than manual check.

Reply:
Thanks!